extensions/amp-story-shopping/0.1/test/test-amp-story-shopping-tag.js

```diff
@@ -28,15 +40,43 @@ describes.realWin(
         'amp-story-shopping-tag',
         {'layout': 'container'}
       );
+
       pageEl.appendChild(element);
       win.document.body.appendChild(pageEl);
 
       shoppingTag = await element.getImpl();
     }
 
-    it('should build shopping tag component', () => {
+    async function shoppingStateHelper() {
+      const shoppingState = {
+        'sunglasses': {'product-title': 'Spectacular Spectacles'},
+      };
+      storeService.dispatch(Action.ADD_SHOPPING_DATA, shoppingState);
+    }
+
+    it('should build and layout shopping tag component', () => {
       expect(() => shoppingTag.layoutCallback()).to.not.throw();
-      expect(shoppingTag.isLayoutSupported(Layout.CONTAINER)).to.be.true;
+    });
+
+    it('should process config data and set text container content if data not null', async () => {
+      shoppingTag.element.setAttribute('data-tag-id', 'sunglasses');
+      await shoppingStateHelper();
+      env.sandbox.stub(shoppingTag, 'mutateElement').callsFake(() => {
+        const shoppingState = {
+          'sunglasses': {'product-title': 'Spectacular Spectacles'},
+        };
+        shoppingTag.element.textContent =
+          shoppingState['sunglasses']['product-title'];
+        expect(shoppingTag.element.textContent).to.equal(
+          'Spectacular Spectacles'
+        );
```

Comment:
This does not test your code as you set the `textContent` yourself here. Shouldn't it be set by the `SHOPPING_DATA` store subscriber in your `amp-story-shopping-tag`?

Reply:
Yes, that's right - these steps aren't necessary.

Reply:
In other words, there is a case where it can be successful, but return null.

Reply:
updated -removed after discussion in standup meeting.